stan/math/rev/functor/adj_jac_apply.hpp

```diff
@@ -272,7 +239,8 @@ struct adj_jac_vari : public vari {
    * recursively)
    */
   template <int R, int C, typename... Pargs>
-  void prepare_x_vis(const Eigen::Matrix<var, R, C>& x, const Pargs&... args) {
+  inline void prepare_x_vis(const Eigen::Matrix<var, R, C>& x,
+                            const Pargs&... args) {
```

Comment:
Would it be feasible in this pull to generalise the Eigen inputs to ```adj_jac_apply``` or should that be a separate PR?

Where this could be something like:
```
  template <typename T, typename... Pargs, require_eigen_st<is_var, T>...>
  void prepare_x_vis(const T& x, const Pargs&... args) {
    static constexpr int t = sizeof...(Targs) - sizeof...(Pargs) - 1;
    Eigen::Map<matrix_vi>(&x_vis_[offsets_[t]], x.rows(), x.cols()) = x.matrix().vi();
    prepare_x_vis(args...);
  }
```

Reply:
> Would it be feasible in this pull to generalise the Eigen inputs

Yeah, separate PR though. Would want to add testing and such. And ideally it should come with a reason for doing it (and function or whatnot we'd be autodiffing).

I'm also curious if there's any forwarding/whatnot we could do to make this match efficiency in situations like the vector ops you were looking at (https://discourse.mc-stan.org/t/adj-jac-apply/5163/6)

Reply:
Cool cool, I'll open an issue and tackle that one after this pull is finished

Reply:
Feel free to think about other problems/opportunities with adj_jac_apply too. Seems like it never caught on like I hoped.

Maybe there are ways to make it simpler, or maybe it's a lack of documentation, I dunno.